### ISSUE TITLE
SELECT on single-byte-charset column (NONE/LATIN1/…) throws 'string right truncation' under default UTF8 encoding

### ISSUE THREAD

**Issue by mariuz** [hgourvest/node-firebird#422]
## Summary

Selecting a text column whose **native charset is single-byte** (`NONE`, `ASCII`, `ISO8859_1`/`LATIN1`, `WIN125x`, …) fails with a server-side **`string right truncation`** error whenever the value is longer than `floor(declaredByteLength / 4)` characters — as long as the connection uses the **default `encoding: 'UTF8'`**.

This makes the stock `employee.fdb` (which is charset `NONE`) unqueryable out of the box:

```
Error: Arithmetic exception, numeric overflow, or string truncation,
       string right truncation, expected length 3, actual 6
```

## Environment

| | |
|---|---|
| node-firebird | 2.11.0 |
| Node.js | 26.4.0 |
| Firebird server | 6.0.0 (also reproducible on 3/4/5) |
| Connection | default options (`encoding` unset ⇒ `UTF8`) |

The bug is independent of the server version; it is a client-side buffer-declaration issue.

## Minimal reproduction

Self-contained — no special database needed, just any DB you can attach to (uses `CAST` to build the offending columns):

```js
const Firebird = require('node-firebird');

Firebird.attach({ host: 'localhost', database: 'employee',
                  user: 'SYSDBA', password: 'masterkey' }, (err, db) => {
  if (err) throw err;
  const run = (sql, label, next) => db.query(sql, (e, r) =>
    (console.log(label, '=>', e ? 'ERROR: ' + e.message : JSON.stringify(r[0])), next && next()));

  // VARCHAR(15) NONE  =>  capacity under UTF8 = floor(15/4) = 3 chars
  run("select cast('abc'  as varchar(15) character set none) v from rdb$database", '3 chars ', () =>
  run("select cast('abcd' as varchar(15) character set none) v from rdb$database", '4 chars ', () =>
  run("select cast('abcd' as varchar(40) character set none) v from rdb$database", '4 in v40', () =>
    db.detach())));
});
```

Output:

```
3 chars  => {"V":"abc"}                                  <- floor(15/4)=3, fits
4 chars  => ERROR: ... string right truncation, expected length 3, actual 4
4 in v40 => {"V":"abcd"}                                 <- floor(40/4)=10, fits
```

The boundary is exactly `floor(byteLength / 4)` characters, which is the tell-tale of a UTF-8 (max 4 bytes/char) capacity calculation being applied to a single-byte column.

## Actual vs. expected

- **Actual:** a hard error is thrown for perfectly valid rows; any `NONE`/single-byte column with more than `len/4` characters is unreadable.
- **Expected:** the value is returned (`"abcd"`, `"Robert"`, …), as it is when the column's native charset is `UTF8`, or when the client sets `encoding: 'NONE'`.

## Root cause

1. On attach, the driver sets the connection character set to the JS `encoding` option, defaulting to **UTF8** (`lib/wire/connection.js:659`):

   ```js
   blr.addString(const.isc_dpb_lc_ctype, options.encoding || 'UTF8', const.DEFAULT_ENCODING);
   ```

2. For a text output column, `SQLVarString.calcBlr` / `SQLVarText.calcBlr` declare the field to the engine using the **server-reported native byte length** (`isc_info_sql_length`) unchanged — e.g. `blr_varying, 15` for `VARCHAR(15) NONE` (confirmed by instrumenting `calcBlr`: it emits `length = 15`).

3. Because the connection charset is UTF8, the engine treats that 15-byte target as holding at most `floor(15 / 4) = 3` **characters** and raises `isc_arith_except / string right truncation (expected length 3, actual N)` when transliterating a source value with more than 3 characters into it.

So the declared *byte* length is correct, but it is too small in *character* terms once the engine is transliterating a narrow charset up to UTF-8. Any charset narrower than UTF-8's 4-byte maximum (i.e. every single-byte charset, plus `UNICODE_FSS`/`SJIS`/`EUCJ` for longer strings) is affected.

Note this is a different failure from #311 (WIN1252 mojibake on display) and #164 (INSERT under NONE): here a **SELECT throws a hard error** rather than returning mis-decoded text.

## Workarounds (for anyone hitting this)

- Set the connection encoding to a single-byte charset that matches the data, e.g. `encoding: 'NONE'` or `'ISO8859_1'` — then width = 1 and the capacity check passes.
- Or `CAST(col AS VARCHAR(n) CHARACTER SET UTF8)` (or a wide `n`) per column — impractical for many columns.

## Suggested fix direction

When the connection charset is wider than a column's native charset, the output buffer must be declared with enough bytes for the engine's character-capacity check to pass — i.e. scale the declared length in `calcBlr` by the ratio of the connection-charset max width to the column-charset width (equivalently, declare `nativeByteLength * connectionCharsetMaxWidth`), and size the receive/decode buffer accordingly. The per-column native charset is already available after `unpackCharSetCollation` (`param.charSetId`), so the scaling factor can be derived without extra round-trips.

**Comment by mariuz** [comment]
Fixed on master (d4197f5, ships in the next release), following exactly the suggested direction: `scaleOutputLengths` runs after describe and widens the declared byte length of text columns whose charset the server did not transliterate (the describe probe confirmed the mechanism — a `WIN1251 VARCHAR(15)` is already rewritten by the server to UTF8/length 60, while `NONE` keeps 15 natively and then trips the fetch-time capacity check). Declared lengths scale by the charset-width ratio (capped at the 16-bit BLR limit), the decoders' width trim restores the true character capacity, and `withMeta` metadata reports the native length.

All repro cases pass: `cast('abcd' as varchar(15) character set none)` returns `"abcd"`, long values and `CHAR` padding are exact, ASCII columns too — and stock `employee.fdb` is queryable out of the box under the default UTF8 connection. Regression tests in `test/codepages.js`.
